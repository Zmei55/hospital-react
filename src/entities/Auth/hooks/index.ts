export * from "./useLogIn";
export * from "./useLogOut";
export * from "./useHandleChange";
