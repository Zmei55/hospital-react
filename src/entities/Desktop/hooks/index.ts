export * from "./useNewRequest";
export * from "./useFindRequest";
