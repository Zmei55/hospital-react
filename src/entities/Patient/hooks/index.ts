export * from "./useChoosePatient";
export * from "./useGetPatient";
export * from "./usePatientsListFind";
export * from "./useHandleInfoActive";
export * from "./useClearPatient";
export * from "./useTogglePatientsModal";
