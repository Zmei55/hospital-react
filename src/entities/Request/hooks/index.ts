export * from "./useRequestServices";
export * from "./useSendRequestDetails";
