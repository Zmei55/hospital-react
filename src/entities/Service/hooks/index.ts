export * from "./useServiceAdd";
export * from "./useServiceDelete";
export * from "./useSelectedServiceDelete";
export * from "./useSelectedServiceListSave";
export * from "./useSelectedServiceListClear";
export * from "./useGetLaborsList";
export * from "./useGetServiceById";
export * from "./useGetServiceListById";
export * from "./useHandleLaborChange";
export * from "./useHandleServicesForm";
export * from "./useToggleLaborModal";
export * from "./useToggleServicesModal";
