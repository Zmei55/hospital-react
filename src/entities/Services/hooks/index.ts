export * from "./useSavedServicesList";
export * from "./useServicesFilter";
export * from "./useAddService";
