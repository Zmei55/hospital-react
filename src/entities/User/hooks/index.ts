export * from "./useGetUserData";
export * from "./useToggleMenuModal";
