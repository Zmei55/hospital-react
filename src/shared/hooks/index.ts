export * from "./useAppSelector";
export * from "./useAppDispatch";
export * from "./useAppNavigate";
export * from "./useToggle";
export * from "./useInput";
