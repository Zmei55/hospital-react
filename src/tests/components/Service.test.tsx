import { render, screen, cleanup } from "@testing-library/react";
import {
  allTheProviders,
  patientMocks,
  serviceListMocks,
  selectedServiceListMocks,
  laborListMocks,
  detailListMocks,
} from "tests";
import * as sharedHooks from "shared/hooks";
import * as serviceHooks from "entities/Service/hooks";
import { ServicesEl } from "entities/Service";

afterEach(() => {
  cleanup();
});

jest.mock("shared/hooks");
jest.mock("entities/Service/hooks");

const mockedUseSelector = jest.spyOn(sharedHooks, "useAppSelector");
const mockedUseServiceToggle = jest.spyOn(
  serviceHooks,
  "useToggleServicesModal"
);
const mockedUseLaborToggle = jest.spyOn(serviceHooks, "useToggleLaborModal");
const mockedUseHandleServicesForm = jest.spyOn(
  serviceHooks,
  "useHandleServicesForm"
);
const mockedUseServiceAdd = jest.spyOn(serviceHooks, "useServiceAdd");
const mockedUseServiceDelete = jest.spyOn(serviceHooks, "useServiceDelete");
const mockedUseSelectedServiceDelete = jest.spyOn(
  serviceHooks,
  "useSelectedServiceDelete"
);
const mockedUseSelectedServiceListSave = jest.spyOn(
  serviceHooks,
  "useSelectedServiceListSave"
);
const mockedUseSelectedServiceListClear = jest.spyOn(
  serviceHooks,
  "useSelectedServiceListClear"
);
const mockedUseHandleLaborChange = jest.spyOn(
  serviceHooks,
  "useHandleLaborChange"
);

describe("Services component", () => {
  it("Services renders", () => {
    mockedUseServiceToggle.mockReturnValue({
      showServiceModal: false,
      toggleServicesModal: jest.fn(),
    });
    mockedUseLaborToggle.mockReturnValue({
      showLaborModal: false,
      toggleLaborModal: jest.fn(),
    });

    render(allTheProviders(<ServicesEl />));

    expect(screen.getByTestId("services-container")).toBeInTheDocument();
  });
});

describe("Services Search component", () => {
  beforeEach(() => {
    mockedUseServiceAdd.mockReturnValue({ addService: jest.fn() });
    mockedUseSelectedServiceDelete.mockReturnValue({
      removeSelectedService: jest.fn(),
    });
    mockedUseSelectedServiceListSave.mockReturnValue({
      saveSelectedList: jest.fn(),
    });
    mockedUseSelectedServiceListClear.mockReturnValue({
      clearSelectedList: jest.fn(),
    });
  });

  it("Services Search renders (without header-buttons-box)", () => {
    mockedUseServiceToggle.mockReturnValue({
      showServiceModal: false,
      toggleServicesModal: jest.fn(),
    });
    mockedUseLaborToggle.mockReturnValue({
      showLaborModal: false,
      toggleLaborModal: jest.fn(),
    });

    const { container } = render(allTheProviders(<ServicesEl />));

    expect(screen.getByTestId("services-container")).toBeInTheDocument();
    expect(screen.getByTestId("services-header")).toBeInTheDocument();
    expect(screen.getByText("Services")).toBeInTheDocument();
    expect(screen.getByTestId("add-services-btn")).toBeInTheDocument();
    expect(
      screen.getByText("Services can be added after selecting a patient")
    ).toBeInTheDocument();
    expect(
      screen.queryByTestId("services-header-buttons-box")
    ).not.toBeInTheDocument();

    expect(container).toMatchSnapshot();
  });

  describe("Services Search-modal component", () => {
    beforeEach(() => {
      mockedUseServiceToggle.mockReturnValue({
        showServiceModal: true,
        toggleServicesModal: jest.fn(),
      });
      mockedUseLaborToggle.mockReturnValue({
        showLaborModal: false,
        toggleLaborModal: jest.fn(),
      });
    });

    it("Services Search-modal renders", () => {
      const stateSelectors = {
        patients: {
          patient: patientMocks,
        },
        services: {
          services: null,
          labors: null,
          details: null,
          selectedServices: null,
        },
      };
      // @ts-ignore
      mockedUseSelector.mockImplementation(state => state(stateSelectors));
      mockedUseHandleServicesForm.mockReturnValue({
        handleServicesForm: jest.fn(),
        filteredServicesList: null,
        isLoading: false,
        isError: false,
      });

      render(allTheProviders(<ServicesEl />));

      expect(screen.getByTestId("services-container")).toBeInTheDocument();
      expect(screen.getByTestId("backdrop")).toBeInTheDocument();
      expect(screen.getByTestId("modal-content")).toBeInTheDocument();
      expect(screen.getByTestId("services-modal-header")).toBeInTheDocument();
      expect(
        screen.queryByTestId("services-header-buttons-box")
      ).not.toBeInTheDocument();
      expect(screen.getByText("Add service")).toBeInTheDocument();
      expect(
        screen.getByTestId("close-modal-services-btn")
      ).toBeInTheDocument();
      expect(screen.getByTestId("services-modal-body")).toBeInTheDocument();
      expect(screen.getByTestId("services-modal-filter")).toBeInTheDocument();
      expect(
        screen.getByTestId("services-modal-filter-form")
      ).toBeInTheDocument();
      expect(
        screen.getByTestId("services-modal-filter-input")
      ).toBeInTheDocument();
      expect(screen.getByTestId("find-service-btn")).toBeInTheDocument();
      expect(screen.queryByTestId("spinner")).not.toBeInTheDocument();
      expect(screen.getByTestId("find-service-btn-text")).toBeInTheDocument();
      expect(screen.getByTestId("services-modal-selected")).toBeInTheDocument();
      expect(
        screen.getByTestId("services-modal-selected-title")
      ).toBeInTheDocument();
      expect(screen.getByText("Selected services")).toBeInTheDocument();
      expect(screen.getByTestId("services-modal-buttons")).toBeInTheDocument();
      expect(
        screen.getByTestId("reset-selected-services-btn")
      ).toBeInTheDocument();
      expect(
        screen.getByTestId("save-selected-services-btn")
      ).toBeInTheDocument();
    });

    it("Services Search-modal renders with spinner loading filtered list", () => {
      const stateSelectors = {
        patients: {
          patient: patientMocks,
        },
        services: {
          services: null,
          labors: null,
          details: null,
          selectedServices: null,
        },
      };
      // @ts-ignore
      mockedUseSelector.mockImplementation(state => state(stateSelectors));
      mockedUseHandleServicesForm.mockReturnValue({
        handleServicesForm: jest.fn(),
        filteredServicesList: null,
        isLoading: true,
        isError: false,
      });

      render(allTheProviders(<ServicesEl />));

      expect(screen.getByTestId("services-container")).toBeInTheDocument();
      expect(screen.getByTestId("services-modal-filter")).toBeInTheDocument();
      expect(screen.getByTestId("find-service-btn")).toBeInTheDocument();
      expect(
        screen.queryByTestId("find-service-btn-text")
      ).not.toBeInTheDocument();
      expect(screen.getByTestId("spinner")).toBeInTheDocument();
    });

    it("Services Search-modal renders with filtered list", () => {
      const stateSelectors = {
        patients: {
          patient: patientMocks,
        },
        services: {
          services: null,
          labors: null,
          details: null,
          selectedServices: null,
        },
      };
      // @ts-ignore
      mockedUseSelector.mockImplementation(state => state(stateSelectors));
      mockedUseHandleServicesForm.mockReturnValue({
        handleServicesForm: jest.fn(),
        filteredServicesList: serviceListMocks,
        isLoading: false,
        isError: false,
      });

      const { container } = render(allTheProviders(<ServicesEl />));

      expect(screen.getByTestId("services-container")).toBeInTheDocument();
      expect(screen.getByTestId("services-modal-filter")).toBeInTheDocument();
      expect(
        screen.getByTestId("services-modal-filter-list")
      ).toBeInTheDocument();

      const listitem = screen.getAllByTestId(
        "services-modal-filter-listitem"
      )[0] as HTMLAnchorElement;
      expect(listitem).toBeInTheDocument();

      const nameBox = screen.getAllByTestId(
        "services-modal-filter-name-box"
      )[0] as HTMLAnchorElement;
      expect(nameBox).toBeInTheDocument();

      const code = screen.getAllByTestId(
        "services-modal-filter-code"
      )[0] as HTMLAnchorElement;
      expect(code).toBeInTheDocument();

      const name = screen.getAllByTestId(
        "services-modal-filter-name"
      )[0] as HTMLAnchorElement;
      expect(name).toBeInTheDocument();

      const addBtn = screen.getAllByTestId(
        "add-service-btn"
      )[0] as HTMLAnchorElement;
      expect(addBtn).toBeInTheDocument();

      const addText = screen.getAllByText("Add")[0] as HTMLAnchorElement;
      expect(addText).toBeInTheDocument();

      expect(container).toMatchSnapshot();
    });

    it("Services Search-modal renders with selected list", () => {
      const stateSelectors = {
        patients: {
          patient: patientMocks,
        },
        services: {
          services: null,
          labors: null,
          details: null,
          selectedServices: selectedServiceListMocks,
        },
      };
      // @ts-ignore
      mockedUseSelector.mockImplementation(state => state(stateSelectors));
      mockedUseHandleServicesForm.mockReturnValue({
        handleServicesForm: jest.fn(),
        filteredServicesList: serviceListMocks,
        isLoading: false,
        isError: false,
      });

      const { container } = render(allTheProviders(<ServicesEl />));

      expect(screen.getByTestId("services-container")).toBeInTheDocument();
      expect(screen.getByTestId("services-modal-selected")).toBeInTheDocument();
      expect(
        screen.getByTestId("services-modal-selected-title")
      ).toBeInTheDocument();
      expect(
        screen.getByTestId("services-modal-selected-list")
      ).toBeInTheDocument();

      const listitem = screen.getAllByTestId(
        "services-modal-selected-listitem"
      )[0] as HTMLAnchorElement;
      expect(listitem).toBeInTheDocument();

      const deleteBtn = screen.getAllByTestId(
        "delete-selected-service-btn"
      )[0] as HTMLAnchorElement;
      expect(deleteBtn).toBeInTheDocument();

      const nameBox = screen.getAllByTestId(
        "services-modal-selected-name-box"
      )[0] as HTMLAnchorElement;
      expect(nameBox).toBeInTheDocument();

      const code = screen.getAllByTestId(
        "services-modal-selected-code"
      )[0] as HTMLAnchorElement;
      expect(code).toBeInTheDocument();

      const name = screen.getAllByTestId(
        "services-modal-selected-name"
      )[0] as HTMLAnchorElement;
      expect(name).toBeInTheDocument();

      expect(screen.getByText("Enzyme")).toBeInTheDocument();

      expect(screen.getByTestId("services-modal-buttons")).toBeInTheDocument();
      expect(
        screen.getByTestId("reset-selected-services-btn")
      ).toBeInTheDocument();
      expect(screen.getByText("Abort")).toBeInTheDocument();
      expect(
        screen.getByTestId("save-selected-services-btn")
      ).toBeInTheDocument();
      expect(screen.getByText("Save and close")).toBeInTheDocument();

      expect(container).toMatchSnapshot();
    });
  });

  describe("Services Labor-modal component", () => {
    beforeEach(() => {
      mockedUseServiceToggle.mockReturnValue({
        showServiceModal: false,
        toggleServicesModal: jest.fn(),
      });
      mockedUseLaborToggle.mockReturnValue({
        showLaborModal: true,
        toggleLaborModal: jest.fn(),
      });
    });

    it("Services Labor-modal renders", () => {
      const stateSelectors = {
        patients: {
          patient: patientMocks,
        },
        services: {
          services: selectedServiceListMocks,
          labors: laborListMocks,
          details: null,
          selectedServices: null,
        },
      };
      // @ts-ignore
      mockedUseSelector.mockImplementation(state => state(stateSelectors));
      mockedUseHandleServicesForm.mockReturnValue({
        handleServicesForm: jest.fn(),
        filteredServicesList: serviceListMocks,
        isLoading: false,
        isError: false,
      });
      mockedUseHandleLaborChange.mockReturnValue({
        handleLaborChange: jest.fn(),
        saveDetails: jest.fn(),
        detailsState: null,
      });
      mockedUseServiceDelete.mockReturnValue({ removeService: jest.fn() });

      const { container } = render(allTheProviders(<ServicesEl />));

      expect(screen.getByTestId("services-container")).toBeInTheDocument();
      expect(screen.getByTestId("modal-content")).toBeInTheDocument();
      expect(screen.getByTestId("labors-modal-header")).toBeInTheDocument();
      expect(
        screen.getByTestId("labors-modal-header-title")
      ).toBeInTheDocument();
      expect(screen.getByTestId("close-labor-modal-btn")).toBeInTheDocument();
      expect(screen.getByTestId("labors-modal-form")).toBeInTheDocument();
      expect(screen.getByTestId("labors-modal-table")).toBeInTheDocument();
      expect(screen.getByTestId("labors-modal-table-head")).toBeInTheDocument();
      expect(screen.getByText("Service")).toBeInTheDocument();

      const laborName = screen.getAllByTestId(
        "labors-table-head-name"
      )[0] as HTMLAnchorElement;
      expect(laborName).toBeInTheDocument();

      expect(screen.getByTestId("labors-modal-table-body")).toBeInTheDocument();

      const tableRow = screen.getAllByTestId(
        "labors-table-body-row"
      )[0] as HTMLAnchorElement;
      expect(tableRow).toBeInTheDocument();

      const serviceBox = screen.getAllByTestId(
        "labors-table-body-service-box"
      )[0] as HTMLAnchorElement;
      expect(serviceBox).toBeInTheDocument();

      const serviceCode = screen.getAllByTestId(
        "labors-table-body-service-code"
      )[0] as HTMLAnchorElement;
      expect(serviceCode).toBeInTheDocument();

      const serviceName = screen.getAllByTestId(
        "labors-table-body-service-name"
      )[0] as HTMLAnchorElement;
      expect(serviceName).toBeInTheDocument();

      const radioButton = screen.getAllByTestId(
        "labors-table-body-radio-button"
      )[0] as HTMLAnchorElement;
      expect(radioButton).toBeInTheDocument();

      const radioInput = screen.getAllByTestId(
        "labors-table-body-radio-input"
      )[0] as HTMLAnchorElement;
      expect(radioInput).toBeInTheDocument();

      expect(
        screen.getByTestId("save-selected-labors-btn")
      ).toBeInTheDocument();
      expect(screen.getByText("Save and close")).toBeInTheDocument();

      expect(container).toMatchSnapshot();
    });
  });

  describe("Services Info component", () => {
    beforeEach(() => {
      mockedUseServiceToggle.mockReturnValue({
        showServiceModal: false,
        toggleServicesModal: jest.fn(),
      });
      mockedUseLaborToggle.mockReturnValue({
        showLaborModal: false,
        toggleLaborModal: jest.fn(),
      });
    });

    it("Services Labor-modal renders", () => {
      const stateSelectors = {
        patients: {
          patient: patientMocks,
        },
        services: {
          services: selectedServiceListMocks,
          labors: laborListMocks,
          details: detailListMocks,
          selectedServices: null,
        },
      };
      // @ts-ignore
      mockedUseSelector.mockImplementation(state => state(stateSelectors));
      mockedUseHandleServicesForm.mockReturnValue({
        handleServicesForm: jest.fn(),
        filteredServicesList: serviceListMocks,
        isLoading: false,
        isError: false,
      });
      mockedUseHandleLaborChange.mockReturnValue({
        handleLaborChange: jest.fn(),
        saveDetails: jest.fn(),
        detailsState: null,
      });
      mockedUseServiceDelete.mockReturnValue({ removeService: jest.fn() });

      const { container } = render(allTheProviders(<ServicesEl />));

      expect(screen.getByTestId("services-container")).toBeInTheDocument();
      expect(screen.getByTestId("services-header")).toBeInTheDocument();
      expect(
        screen.getByTestId("services-header-buttons-box")
      ).toBeInTheDocument();
      expect(
        screen.getByTestId("add-to-selected-service-btn")
      ).toBeInTheDocument();
      expect(screen.getByTestId("select-labor")).toBeInTheDocument();
      expect(screen.getByTestId("services-info-list")).toBeInTheDocument();

      const listitem = screen.getAllByTestId(
        "services-info-listitem"
      )[0] as HTMLAnchorElement;
      expect(listitem).toBeInTheDocument();

      const deleteService = screen.getAllByTestId(
        "delete-service-info-btn"
      )[0] as HTMLAnchorElement;
      expect(deleteService).toBeInTheDocument();

      const infoBox = screen.getAllByTestId(
        "services-info-box"
      )[0] as HTMLAnchorElement;
      expect(infoBox).toBeInTheDocument();

      const boxHeader = screen.getAllByTestId(
        "services-info-box-header"
      )[0] as HTMLAnchorElement;
      expect(boxHeader).toBeInTheDocument();

      const serviceCode = screen.getAllByTestId(
        "services-info-box-service-code"
      )[0] as HTMLAnchorElement;
      expect(serviceCode).toBeInTheDocument();

      const serviceName = screen.getAllByTestId(
        "services-info-box-service-name"
      )[0] as HTMLAnchorElement;
      expect(serviceName).toBeInTheDocument();

      expect(container).toMatchSnapshot();
    });
  });
});
