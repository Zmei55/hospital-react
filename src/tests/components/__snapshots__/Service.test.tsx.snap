// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Services Search component Services Info component Services Labor-modal renders 1`] = `
<div>
  <div
    class="css-185l0bc"
    data-testid="services-container"
  >
    <div
      class="css-133fwc"
      data-testid="services-header"
    >
      <h2
        class="css-1wru4hq"
      >
        Services
      </h2>
      <div
        class="css-ov1ktg"
        data-testid="services-header-buttons-box"
      >
        <button
          class="css-bewdgy"
          data-testid="add-to-selected-service-btn"
          style="height: 48px; padding-right: 24px; padding-left: 24px; margin-right: 24px;"
        >
          Add
        </button>
        <button
          class="css-bewdgy"
          data-testid="select-labor"
          style="height: 48px; padding-right: 24px; padding-left: 24px;"
        >
          Choice of the lab
        </button>
      </div>
    </div>
    <ul
      class="css-1edcxm"
      data-testid="services-info-list"
    >
      <li
        class="css-e10vzf"
        data-testid="services-info-listitem"
      >
        <button
          class="css-ch4pr2"
          data-testid="delete-service-info-btn"
          style="width: 52px; height: 52px; margin-right: 24px;"
        >
          <svg
            class="css-nhqhda"
            fill="red"
            height="48"
            width="48"
          >
            <use
              href="svgSprite.svg#trash"
            />
          </svg>
        </button>
        <div
          class="css-8tpwv0"
          data-testid="services-info-box"
        >
          <div
            class="css-g11d5q"
            data-testid="services-info-box-header"
          >
            <span
              class="css-1ixeebt"
              data-testid="services-info-box-service-code"
            >
              L18.36.00.0.005
            </span>
          </div>
          <span
            class="css-xb338v"
            data-testid="services-info-box-service-name"
          >
            Enzyme
          </span>
        </div>
      </li>
      <li
        class="css-e10vzf"
        data-testid="services-info-listitem"
      >
        <button
          class="css-ch4pr2"
          data-testid="delete-service-info-btn"
          style="width: 52px; height: 52px; margin-right: 24px;"
        >
          <svg
            class="css-nhqhda"
            fill="red"
            height="48"
            width="48"
          >
            <use
              href="svgSprite.svg#trash"
            />
          </svg>
        </button>
        <div
          class="css-8tpwv0"
          data-testid="services-info-box"
        >
          <div
            class="css-g11d5q"
            data-testid="services-info-box-header"
          >
            <span
              class="css-1ixeebt"
              data-testid="services-info-box-service-code"
            >
              L18.36.00.0.006
            </span>
          </div>
          <span
            class="css-xb338v"
            data-testid="services-info-box-service-name"
          >
            Gerinnung
          </span>
        </div>
      </li>
    </ul>
  </div>
</div>
`;

exports[`Services Search component Services Labor-modal component Services Labor-modal renders 1`] = `
<div>
  <div
    class="css-185l0bc"
    data-testid="services-container"
  >
    <div
      class="css-133fwc"
      data-testid="services-header"
    >
      <h2
        class="css-1wru4hq"
      >
        Services
      </h2>
      <div
        class="css-ov1ktg"
        data-testid="services-header-buttons-box"
      >
        <button
          class="css-bewdgy"
          data-testid="add-to-selected-service-btn"
          style="height: 48px; padding-right: 24px; padding-left: 24px; margin-right: 24px;"
        >
          Add
        </button>
        <button
          class="css-bewdgy"
          data-testid="select-labor"
          style="height: 48px; padding-right: 24px; padding-left: 24px;"
        >
          Choice of the lab
        </button>
      </div>
    </div>
    <ul
      class="css-1edcxm"
      data-testid="services-info-list"
    >
      <li
        class="css-e10vzf"
        data-testid="services-info-listitem"
      >
        <button
          class="css-ch4pr2"
          data-testid="delete-service-info-btn"
          style="width: 52px; height: 52px; margin-right: 24px;"
        >
          <svg
            class="css-nhqhda"
            fill="red"
            height="48"
            width="48"
          >
            <use
              href="svgSprite.svg#trash"
            />
          </svg>
        </button>
        <div
          class="css-8tpwv0"
          data-testid="services-info-box"
        >
          <div
            class="css-g11d5q"
            data-testid="services-info-box-header"
          >
            <span
              class="css-1ixeebt"
              data-testid="services-info-box-service-code"
            >
              L18.36.00.0.005
            </span>
          </div>
          <span
            class="css-xb338v"
            data-testid="services-info-box-service-name"
          >
            Enzyme
          </span>
        </div>
      </li>
      <li
        class="css-e10vzf"
        data-testid="services-info-listitem"
      >
        <button
          class="css-ch4pr2"
          data-testid="delete-service-info-btn"
          style="width: 52px; height: 52px; margin-right: 24px;"
        >
          <svg
            class="css-nhqhda"
            fill="red"
            height="48"
            width="48"
          >
            <use
              href="svgSprite.svg#trash"
            />
          </svg>
        </button>
        <div
          class="css-8tpwv0"
          data-testid="services-info-box"
        >
          <div
            class="css-g11d5q"
            data-testid="services-info-box-header"
          >
            <span
              class="css-1ixeebt"
              data-testid="services-info-box-service-code"
            >
              L18.36.00.0.006
            </span>
          </div>
          <span
            class="css-xb338v"
            data-testid="services-info-box-service-name"
          >
            Gerinnung
          </span>
        </div>
      </li>
    </ul>
  </div>
</div>
`;

exports[`Services Search component Services Search renders (without header-buttons-box) 1`] = `
<div>
  <div
    class="css-185l0bc"
    data-testid="services-container"
  >
    <div
      class="css-133fwc"
      data-testid="services-header"
    >
      <h2
        class="css-1wru4hq"
      >
        Services
      </h2>
    </div>
    <button
      class="css-na7bfu"
      data-testid="add-services-btn"
      disabled=""
      type="button"
    >
      <svg
        class="css-smt8v5"
        height="48"
        width="48"
      >
        <use
          href="svgSprite.svg#list-plus"
        />
      </svg>
      <span
        class="css-m693jn"
      >
        Services can be added after selecting a patient
      </span>
    </button>
  </div>
</div>
`;

exports[`Services Search component Services Search-modal component Services Search-modal renders with filtered list 1`] = `
<div>
  <div
    class="css-185l0bc"
    data-testid="services-container"
  >
    <div
      class="css-133fwc"
      data-testid="services-header"
    >
      <h2
        class="css-1wru4hq"
      >
        Services
      </h2>
    </div>
    <button
      class="css-na7bfu"
      data-testid="add-services-btn"
      type="button"
    >
      <svg
        class="css-smt8v5"
        height="48"
        width="48"
      >
        <use
          href="svgSprite.svg#list-plus"
        />
      </svg>
      <span
        class="css-m693jn"
      >
        Add services
      </span>
    </button>
  </div>
</div>
`;

exports[`Services Search component Services Search-modal component Services Search-modal renders with selected list 1`] = `
<div>
  <div
    class="css-185l0bc"
    data-testid="services-container"
  >
    <div
      class="css-133fwc"
      data-testid="services-header"
    >
      <h2
        class="css-1wru4hq"
      >
        Services
      </h2>
    </div>
    <button
      class="css-na7bfu"
      data-testid="add-services-btn"
      type="button"
    >
      <svg
        class="css-smt8v5"
        height="48"
        width="48"
      >
        <use
          href="svgSprite.svg#list-plus"
        />
      </svg>
      <span
        class="css-m693jn"
      >
        Add services
      </span>
    </button>
  </div>
</div>
`;
